crates/test-util/src/actor.rs

```diff
@@ -0,0 +1,103 @@
+
+use wascap::{jwt, wasm::extract_claims};
+use wasmcloud_control_interface::{Client as WasmcloudCtrlClient, CtlOperationAck};
+
+/// Start an actor, ensuring that the actor starts properly
+pub async fn assert_start_actor(
+    ctl_client: impl AsRef<WasmcloudCtrlClient>,
+    lattice: impl AsRef<str>,
+    host_key: impl AsRef<KeyPair>,
+    url: impl AsRef<str>,
+    count: u32,
+) -> Result<()> {
+    let lattice = lattice.as_ref();
+    let ctl_client = ctl_client.as_ref();
+    let nats_client = ctl_client.nats_client();
+    let host_key = host_key.as_ref();
+
+    let subject = format!("wasmbus.evt.{lattice}.actors_started");
+    let mut sub_started = nats_client
+        .subscribe(subject)
+        .await
+        .map_err(|e| anyhow!(e).context("failed to subscribe to NATS subject"))?;
+
+    let CtlOperationAck { accepted, error } = ctl_client
+        .scale_actor(&host_key.public_key(), url.as_ref(), count, None)
+        .await
+        .map_err(|e| anyhow!(e).context("failed to start actor"))?;
+    ensure!(error == "");
+    ensure!(accepted);
+
+    // Naive wait for at least a stopped / started event before exiting this function. This prevents
+    // assuming we're done with scaling too early since scale is an early-ack ctl request.
+    tokio::select! {
+        _ = sub_started.next() => {
+        }
+        _ = tokio::time::sleep(Duration::from_secs(10)) => {
+            bail!("timed out waiting for actor started event");
+        },
+    }
+
+    Ok(())
+}
+
+/// Given a path to an actor on disks, extract claims
+pub async fn extract_actor_claims(
+    wasm_binary_path: impl AsRef<Path>,
+) -> Result<jwt::Claims<jwt::Actor>> {
+    let wasm_binary_path = wasm_binary_path.as_ref();
+    let jwt::Token { claims, .. } = extract_claims(fs::read(wasm_binary_path).await?)
+        .context("failed to extract kv http smithy actor claims")?
+        .context("component actor claims missing")?;
+    Ok(claims)
+}
+
+/// Scale an actor, ensuring that the scale up/down was successful
+pub async fn assert_scale_actor(
+    ctl_client: impl AsRef<wasmcloud_control_interface::Client>,
+    lattice_name: impl AsRef<str>,
+    host_key: impl AsRef<KeyPair>,
+    url: impl AsRef<str>,
+    annotations: Option<HashMap<String, String>>,
+    count: u32,
+) -> anyhow::Result<()> {
+    let host_key = host_key.as_ref();
+    let ctl_client = ctl_client.as_ref();
+    let nats_client = ctl_client.nats_client();
+    let lattice_name = lattice_name.as_ref();
+    let mut sub_started = nats_client
+        .subscribe(format!("wasmbus.evt.{lattice_name}.actors_started"))
+        .await?;
+    let mut sub_stopped = nats_client
+        .subscribe(format!("wasmbus.evt.{lattice_name}.actors_stopped"))
+        .await?;
+    let CtlOperationAck { accepted, error } = ctl_client
+        .scale_actor(&host_key.public_key(), url.as_ref(), count, annotations)
+        .await
+        .map_err(|e| anyhow!(e).context("failed to start actor"))?;
```

Comment:
Now that we have `actor_scaled`, you can actually just assert that we receive that event and that it's the count we asked for!

Reply:
Does the comment below not apply? Waiting for stopped/started instead of scale because of scale being early-ACK? Has how it worked changed?

Reply:
Oh! Yes that still applies. I was just saying that instead of waiting on actors_started or actors_stopped, you can simply wait for actor_scaled and that will contain the count that it was scaled to

We added that new event to cut down on the number of events, how weird it was to calculate new vs old running instances, and now the event is more idempotent

Reply:
Right, but why the code waited for started/stopped was because scale ACKs early... Are you saying we don't need to do that, and I can wait for just `actor_scaled`? If I can, then I'll do that!

Reply:
Are start actor/scale actor broken? start provider works just fine as the previous code, but `event_receiver` utility doesn't seem to work with started or scaled actors.